File: yejee/week4/큰 수 만들기.java
class Solution {
    public String solution(String number, int k) {
        int num_len = number.length(); 
        char max_num;
        int max_i = 0;
        int erasen = 0; //현재까지 지워진 숫자의 개수 
        
        StringBuilder answer = new StringBuilder();
        
        for(int j = 0; j < num_len && answer.length() < num_len - k ; j++){
            max_num = number.charAt(j);
            for(int i = j ; i < j + k - erasen && i < num_len - 1 ; i++){
                if(max_num < number.charAt(i+1)){
                    max_num = number.charAt(i+1);
                    max_i = i+1;
                }
            }
            answer.append(String.valueOf(max_num));
            erasen += max_i - j;
            j = max_i;
            max_i ++;
        }
        return answer.toString();
    }
}

File: SiHyeon/week4/큰 수 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string number, int k) {
    string answer = "";
    for (int i = 1; i < number.size() && k > 0; i++) //k값이 0보다 크고 number를 끝까지 돌기 전까지 순회한다
    {
        if (number[i - 1] < number[i])
        {
            number.erase(i - 1, 1);
            i = 0;
            k--;
        }
    }
    answer = number.erase(number.size() - k, k); //다 돌아도 k값이 남은 경우는 끝에서 빼준다
    return answer;
}